src/directives/layertree.js

```diff
@@ -241,14 +264,93 @@ ngeo.LayertreeController = function($scope, $element, $attrs, ngeoDecorateLayer,
 
 
 /**
+ * Return the current state.
+ * @return {string} 'on', 'off', 'indeterminate'.
+ * @export
+ */
+ngeo.LayertreeController.prototype.getState = function() {
+  return this.state_;
+};
+
+
+/**
+ * Set the state of this treeCtrl. Update its children with its value and then
+ * ask its parent to refresh its state.
+ * @param {string} state 'on' or 'off'.
+ * @param {boolean=} opt_avoidRefreshParents True to avoid refreshing its
+ * parent.
+ * @export
+ */
+ngeo.LayertreeController.prototype.setState = function(state,
```

Comment:
Use a setStateInternal to only change the state.
At the end setState, emit a change event (for the whole branch).